map/bookmark_manager.cpp

```diff
@@ -309,6 +309,10 @@ Track * BookmarkManager::GetTrackForEdit(kml::TrackId trackId)
   if (it == m_tracks.end())
     return nullptr;
 
+  auto const groupId = it->second->GetGroupId();
+  if (groupId != kml::kInvalidTrackId)
```

Comment:
Did you hit the case when groupId was equal to kInvallidTrackId? It would be great to document it.

Reply:
I used the equal approach as used for bookmarks for now and tried not to break and refactor the overall logic:

```cpp
{
Bookmark * BookmarkManager::GetBookmarkForEdit(kml::MarkId markId)
  CHECK_THREAD_CHECKER(m_threadChecker, ());
  auto it = m_bookmarks.find(markId);
  if (it == m_bookmarks.end())
    return nullptr;

  auto const groupId = it->second->GetGroupId();
  if (groupId != kml::kInvalidMarkGroupId)
    m_changesTracker.OnUpdateMark(markId);

  return it->second.get();
}
```